src/main/java/session/FormEntrySession.java

```diff
@@ -83,8 +90,28 @@ public FormEntrySession(String formXml, String restoreXml, String initLang, Stri
         langs = formEntryModel.getLanguages();
         this.initLang = initLang;
         uuid = UUID.randomUUID().toString();
+        this.sequenceId = 1;
+        initialize(true, sessionData);
+        getFormTree();
+    }
+
+    // Entry from menu selection. Assumes user has already been restored.
+    public FormEntrySession(UserSandbox sandbox, FormDef formDef, String initLang, String username,
+                            Map<String, String> sessionData) throws Exception {
+        this.username = username;
+        this.sessionData = sessionData;
+        this.formDef = formDef;
+        this.sandbox = sandbox;
+        formEntryModel = new FormEntryModel(formDef, FormEntryModel.REPEAT_STRUCTURE_NON_LINEAR);
+        formEntryController = new FormEntryController(formEntryModel);
+        formEntryController.setLanguage(initLang);
+        title = formDef.getTitle();
+        langs = formEntryModel.getLanguages();
+        this.initLang = initLang;
+        uuid = UUID.randomUUID().toString();
         this.sequenceId = 0;
         initialize(true, sessionData);
+        getFormTree();
```

Comment:
returns the form tree actually haha, I just need to do this here also to make sure the repeats are expanded.